sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -236,7 +236,32 @@ case class StaticInvoke(
   override def children: Seq[Expression] = arguments
 
   lazy val argClasses = ScalaReflection.expressionJavaClasses(arguments)
-  @transient lazy val method = cls.getDeclaredMethod(functionName, argClasses : _*)
+  @transient lazy val method = {
+    try {
+      cls.getDeclaredMethod(functionName, argClasses: _*)
```

Comment:
btw, `java_method` can support more cases via the same resolution logic?

Reply:
Took a rough look at how `java_method` resolves the calling method. `java_method` only calls Java static method, but `StaticInvoke` covers Scala object method too. `java_method`'s argument types seem only for catalyst primitive data types, so complext type and `ObjectType` is not supported.

Reply:
We may extend `java_method` later.